### DISCUSSION TITLE
How do I dim a qbtn until a table row is selected

### DISCUSSION THREAD

**Discussion by michaeloliveraz** [quasarframework/quasar#12436] [General - Components / Directives / etc] [answered]
I have a table with single select.  I have buttons that I want to visibly disable until a row is selected.  I would prefer to dim or grey the button as well as disable it based on the criteria that a row is selected.

Thanks,

**Comment by metalsadman** [comment]
provide a codepen.

**Reply by michaeloliveraz** [reply]
​<​template​> 
 ​    <​div​ ​class​=​"​q-pa-md​"​> 
 ​      <​div​ ​class​=​"​q-pa-md q-gutter-sm​"​> 
 ​      <​q​-btn​ ​color​=​"​primary​"​ ​icon​=​"​assignment​"​ ​label​=​"​Config​"​ /> 
 ​      <​q​-btn​ ​color​=​"​secondary​"​ ​icon​=​"​schema​"​ ​label​=​"​Schema​"​ /> 
 ​      <​q​-btn​ ​color​=​"​red​"​ ​icon​=​"​schedule​"​ ​label​=​"​Jobs​"​ /> 
 ​      <​q​-btn​ ​color​=​"​pink​"​ ​icon​=​"​route​"​ ​label​=​"​Taxonomy​"​ /> 
 ​      <​br​> 
 ​      <​q​-btn​ ​icon​=​"​launch​"​ ​label​=​"​Test​"​ ​glossy​ ​color​=​"​purple​"​ /> 
 ​      <​q​-btn​ ​icon​=​"​add​"​ ​label​=​"​New Collector​"​ ​glossy​ ​color​=​"​green​"​ /> 
 ​    </​div​> 
 ​    <​q​-table 
 ​      ​title​=​"​Collectors​" 
 ​      :​rows​=​"​rows​" 
 ​      :​columns​=​"​columns​" 
 ​      ​row-key​=​"​name​" 
 ​      ​selection​=​"​single​" 
 ​      ​v-model​:​selected​=​"​selected​" 
 ​    /> 
  
 ​    <​div​ ​class​=​"​q-mt-md​"​> 
 ​      Selected: {{ ​JSON​.​stringify​(selected) }} 
 ​    </​div​> 
 ​  </​div​> 
 ​</​template​>

**Comment by gregadro** [comment, marked as answer]
Based on the value of `selected` you can set the `disable` prop on the buttons (`true` when it is empty). This will already "dim" them, but if that is not enough you can apply them a custom CSS class/style.

[CodePen](https://codepen.io/drofx/pen/Bamdpyx)

**Reply by michaeloliveraz** [reply]
Excellent thanks
